plugins/secretstores/systemd/README.md

```diff
@@ -0,0 +1,248 @@
+manually via the command line! Therefore, `secrets` commands should **not**
+be used with this plugin.
+
+## Usage <!-- @/docs/includes/secret_usage.md -->
+
+Secrets defined by a store are referenced with `@{<store-id>:<secret_key>}`
+the Telegraf configuration. Only certain Telegraf plugins and options of
+support secret stores. To see which plugins and options support
+secrets, see their respective documentation (e.g.
+`plugins/outputs/influxdb/README.md`). If the plugin's README has the
+`Secret-store support` section, it will detail which options support secret
+store usage.
+
+## Configuration
+
+```toml @sample.conf
+# Secret-store to access systemd secrets
+[[secretstores.systemd]]
+  ## Unique identifier for the secretstore.
+  ## This id can later be used in plugins to reference the secrets
+  ## in this secret-store via @{<id>:<secret_key>} (mandatory)
+  id = "my_secretstore"
+
+  ## Path to systemd credentials directory
+  ## This should not be required as systemd indicates this directory
+  ## via the CREDENTIALS_DIRECTORY environment variable.
+  # path = "${CREDENTIALS_DIRECTORY}"
+```
+
+Each Secret provided by systemd will be available as file under
+`${CREDENTIALS_DIRECTORY}/<secret-name>` for the service. You will **not**
+be able to see them as a regular user, not are those files accessible to other
+services.
+
+## Credential management
+
+Most steps here are condensed from the [systemd-creds man-page][systemd-creds]
+and are using this command. Please also check that man-page as the options
+or verbs used here might be outdated for the systemd version you are using.
+
+**Please note**: We are using `/etc/telegraf/credentials` as our storage
+location for encrypted credentials throughout the examples below. This is
+because the Telegraf service file expects credentials to be located there
+by default. Furthermore, we assume the secret-store ID to be set to `syscreds`
+in the examples.
+
+### Setup
+
+If you have not done it already, systemd requires a first-time setup of the
+credential system. If you are planning to use the TPM2 chip of your system
+for protecting the credentials, you should first make sure that it is
+available using
+
+```shell
+# sudo systemd-creds has-tpm2
+partial
+-firmware
++driver
++system
++subsystem
+```
+
+The output should look similar to the above.
+
+Now setup the credentials by creating the root key.
+
+```shell
+# sudo systemd-creds setup
+Credential secret file '/var/lib/systemd/credential.secret' is not located on encrypted media, using anyway.
+4096 byte credentials host key set up.
+```
+
+The warning only appears if you are storing the generated key on an unencrypted
+disk which is not recommended. With this, we are all set to create credentials.
+
+### Creating credentials
+
+After setting up the encryption key we can create a new credential using
+
+```shell
+# echo -n "john-doe-jr" | systemd-creds encrypt - /etc/telegraf/credentials/http_user
```

Comment:
I thought about it and came to the conclusion that it is easier to give a single dir in the service file instead of adding _multiple_ entries containing encrypted credentials. Both is possible and supported but I want to keep the examples clean here...